policykit/policyengine/views.py

```diff
@@ -824,49 +824,177 @@ def document_action_recover(request):
 
     return HttpResponse()
 
-def policy_from_request (request, key_name = 'policy'):
+def policy_from_request(request, key_name = 'policy'):
     policy_id = request.GET.get(key_name)
-
     from policyengine.models import Policy
-
-    # Get source policy object
     try:
         return Policy.objects.get(pk=policy_id)
     except Policy.DoesNotExist:
         raise Http404("Policy does not exist")
 
+def get_policy_by_name(request, key_name = "name"):
+    """
+    Get policy by name.
+    """
+    from policyengine.models import Policy
+    name = request.GET.get(key_name)
+    try:
+        return Policy.objects.get(name=name)
+    except Policy.DoesNotExist:
+        return(None)
+
+def embed_populate_templates(request):
```

Comment:
Yeah, I don't want to be creating one-off policies here. Also the policy engine should have no platform-specific logic. This should be a python script runnable from the command line using manage.py and stored in a platform-specific place.